### PR TITLE
Stops bodybags from locking you inside of a morgue tray

### LINKED ISSUE #31656: Can't resist out of a bodybag in a morgue tray
I assume any double container situation will do this as well

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request

Same as title. (Adds a 30 second do after)

Fixes https://github.com/tgstation/tgstation/issues/31656

## Why It's Good For The Game

Removing the ability to GBJ people with little to no effort is a good thing.

## Changelog
:cl:
fix: A bodybag will no longer permanently lock you inside of a morgue tray.
/:cl:

### CODE REVIEW

**Comment by Zytolg** [comment]
It would be neat if you had to struggle out of this situation. Check to see if bodybagged is stuck in a morgue tray, and if they are, start a short countdown (as they struggle to "worm" the morgue tray open). I agree that body bags shouldn't lock you into a morgue tray, but it also doesn't make sense that it'd be easy to get out of that scenario either.

File: code/game/objects/structures/crates_lockers/closets/bodybag.dm [hunk 1]
```diff
@@ -65,6 +65,13 @@
 		density = FALSE
 		mouse_drag_pointer = MOUSE_ACTIVE_POINTER
 
+/obj/structure/closet/body_bag/container_resist_act(mob/living/user)
+	if (istype(src.loc, /obj/structure/bodycontainer))
+		var/obj/structure/bodycontainer/tray = src.loc
+		to_chat(user, "<span class='notice'>You attempt to breakout of the [tray]... (This will take around 30 seconds.)</span>")
+		if(do_after(user, 30 SECONDS))
+			tray.open()
```
Comment (tralezab) [inline code/game/objects/structures/crates_lockers/closets/bodybag.dm:73]: is ?. more appropriate here?

**Comment by LemonInTheDark** [comment]
does this solve all other double container situations? this seems like something that should be more general rather then just bodybags and morgue trays

**Comment by SuperNovaa41** [comment]
Just going to close this until I can figure out a solution

**Pull request closed without merging**